ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -57,18 +59,32 @@
   private final TransactionBatchAddedListener transactionBatchAddedListener;
   private final SyncState syncState;
   private Optional<AccountFilter> accountFilter = Optional.empty();
+  private final PeerTransactionTracker peerTransactionTracker;
 
   public TransactionPool(
       final PendingTransactions pendingTransactions,
       final ProtocolSchedule<?> protocolSchedule,
       final ProtocolContext<?> protocolContext,
       final TransactionBatchAddedListener transactionBatchAddedListener,
-      final SyncState syncState) {
+      final SyncState syncState,
+      final EthContext ethContext,
+      final PeerTransactionTracker peerTransactionTracker) {
     this.pendingTransactions = pendingTransactions;
     this.protocolSchedule = protocolSchedule;
     this.protocolContext = protocolContext;
     this.transactionBatchAddedListener = transactionBatchAddedListener;
     this.syncState = syncState;
+    this.peerTransactionTracker = peerTransactionTracker;
+
+    ethContext.getEthPeers().subscribeConnect(this::handleConnect);
+  }
+
+  private void handleConnect(final EthPeer peer) {
+    List<Transaction> localTransactions = getLocalTransactions();
+    for (Transaction transaction : localTransactions) {
+      peerTransactionTracker.addToPeerSendQueue(peer, transaction);
+    }
+    peerTransactionTracker.markTransactionsAsSeen(peer, localTransactions);
```

Comment:
I don't think you need to call `markTransactionsAsSeen`.  Marking seen transactions will be handled by `TransactionsMessageSender`.